modules/eslint-plugin/src/rule-creator.ts

```diff
@@ -1,69 +1,40 @@
-import type { TSESLint } from '@typescript-eslint/utils';
 import { ESLintUtils } from '@typescript-eslint/utils';
-import type { NGRX_MODULE } from './utils';
-import { docsUrl, ngrxVersionSatisfies, NGRX_MODULE_PATHS } from './utils';
+import { NGRX_MODULE } from './utils';
 
-type Meta<TMessageIds extends string, TOptions extends readonly unknown[]> =
-  | TSESLint.RuleMetaData<TMessageIds, TOptions> & {
-      ngrxModule: NGRX_MODULE;
-      version?: string;
-      docs: { requiresTypeChecking?: boolean };
-    };
-
-type CreateRuleMeta<
-  TMessageIds extends string,
-  TOptions extends readonly unknown[]
-> = {
-  docs: Omit<TSESLint.RuleMetaDataDocs, 'url'> & {
-    requiresTypeChecking?: boolean;
-  };
-} & Omit<Meta<TMessageIds, TOptions>, 'docs'>;
-export type NgRxRuleModule<
-  TOptions extends readonly unknown[],
-  TMessageIds extends string
-> = Omit<TSESLint.RuleModule<TMessageIds, TOptions>, 'meta'> & {
-  meta: Meta<TMessageIds, TOptions>;
-};
+export interface NgRxRuleDocs {
+  ngrxModule: NGRX_MODULE;
+  requiresTypeChecking?: boolean;
+}
 
-export function createRule<
-  TOptions extends readonly unknown[],
-  TMessageIds extends string
->(
-  config: Readonly<{
-    name: string;
-    meta: CreateRuleMeta<TMessageIds, TOptions>;
-    defaultOptions: Readonly<TOptions>;
-    create: (
-      context: Readonly<TSESLint.RuleContext<TMessageIds, TOptions>>,
-      optionsWithDefault: Readonly<TOptions>
-    ) => TSESLint.RuleListener;
-  }>
-): TSESLint.RuleModule<TMessageIds, TOptions> {
-  const configOverwrite = {
-    ...config,
-    create: (
-      context: Readonly<TSESLint.RuleContext<TMessageIds, TOptions>>,
-      optionsWithDefault: Readonly<TOptions>
-    ) => {
-      const {
-        meta: { ngrxModule, version },
-      } = config;
-      if (
-        version !== undefined &&
-        !ngrxVersionSatisfies(NGRX_MODULE_PATHS[ngrxModule], version)
```

Comment:
Everything about the version checks (to disable/enable specific  rules for a particular NgRx version) were removed in the past, but the code was not cleaned up. All rules are always applicable for all versions.
I noticed this only now, because of the TSESlint changes in the rule creator.